src/map/guild.cpp

```diff
@@ -269,25 +270,72 @@ bool guild_check_skill_require( struct guild *g, uint16 id ){
 	return true;
 }
 
-static bool guild_read_castledb(char* str[], int columns, int current) {// <castle id>,<map name>,<castle name>,<castle event>[,<reserved/unused switch flag>]
-	struct guild_castle *gc;
-	int mapindex = mapindex_name2id(str[1]);
-	const int castle_id = atoi(str[0]);
-	
-	if (map_mapindex2mapid(mapindex) < 0) // Map not found or on another map-server
-		return false;
+const std::string CastleDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/castle_db.yml";
+}
+
+uint64 CastleDatabase::parseBodyNode(const YAML::Node &node) {
+	int32 castle_id;
 
-	if ((gc = static_cast<guild_castle*>(idb_get(castle_db, castle_id))) == nullptr) {
-		CREATE(gc, struct guild_castle, 1);
+	if (!this->asInt32(node, "Id", castle_id))
+		return 0;
+
+	std::shared_ptr<guild_castle> gc = this->find(castle_id);
+	bool exists = gc != nullptr;
+
+	if (!exists) {
+		if (!this->nodesExist(node, { "Map", "Name", "NPC" }))
+			return 0;
+
+		gc = std::make_shared<guild_castle>();
+		gc->castle_id = castle_id;
 	}
+
+	if (this->nodeExists(node, "Map")) {
+		std::string map_name;
+
+		if (!this->asString(node, "Map", map_name))
+			return 0;
+
+		uint16 mapindex = mapindex_name2id(map_name.c_str());
```

Comment:
By using this function missing maps are already getting reported. For a silent call you need to call it with mapindex_name2idx(mapname, nullptr)

Reply:
Also this function returns the mapid and not the mapindex. So variable naming is bad or maybe even wrong logic?